src/cdk/a11y/focus-trap/focus-trap.ts

```diff
@@ -171,9 +174,11 @@ export class FocusTrap {
     const redirectToElement = this._element.querySelector(`[cdk-focus-initial], ` +
                                                           `[cdkFocusInitial]`) as HTMLElement;
 
-    if (this._element.hasAttribute(`cdk-focus-initial`)) {
-      console.warn(`Found use of deprecated attribute 'cdk-focus-initial',` +
-                    ` use 'cdkFocusInitial' instead.`, this._element);
+    // @deletion-target 7.0.0
+    if (redirectToElement.hasAttribute(`cdk-focus-initial`)) {
```

Comment:
We are not checking to confirm that `redirectToElement` is not null here.  We are seeing some failing tests, where the querySelector does not find anything which results in  `TypeError: Cannot read property 'hasAttribute' of null`